### DISCUSSION TITLE
It stucked when download `arrow_ep`

### DISCUSSION THREAD

**Discussion by Thespica** [apache/incubator-graphar#188] [Q&A] [answered]
# Problem

When I ran `make` in my build diectory, it just stucked here like this:

```shell
[  8%] Performing download step (download, verify and extract) for 'arrow_ep'
-- Downloading...
   dst='/home/john/ffi-test/graphar-src/cpp/build/arrow_ep-prefix/src/apache-arrow-10.0.1.tar.gz'
   timeout='none'
-- Using src='https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-10.0.1/apache-arrow-10.0.1.ta
r.gz'
-- Retrying...
-- Using src='https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-10.0.1/apache-arrow-10.0.1.ta
r.gz'
-- Retry after 5 seconds (attempt #2) ...
-- Using src='https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-10.0.1/apache-arrow-10.0.1.ta
r.gz'

```

# Environment

WSL Ubuntu 20.04

curl info:

```shell
curl 8.1.2 (x86_64-pc-linux-gnu) libcurl/8.1.2 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 zstd/1.4.4 nghttp2/1.40.0
Release-Date: 2023-05-30
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tft
p
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTPS-proxy IPv6 Largefile libz NTLM NTLM_WB SSL threadsafe TLS-SRP Unix
Sockets zstd

```

Apache arrow version: 12.0.0, installed by apt.

Is there any solution for that, or maybe I should reinstall arrow from source?

**Comment by acezen** [comment, marked as answer]
GraphAr would download arrow source code and build, link as static library and it does not link to the system installed arrow.
The problem seems that  your network is slow to download https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-10.0.1/apache-arrow-10.0.1.tar.gz with curl. Try to download manually and put it to `/home/john/ffi-test/graphar-src/cpp/build/arrow_ep-prefix/src/ ` and build again.

**Reply by Thespica** [reply]
Thanks, no problem here.

`make` worked successfully with downloading arrow 10.0.1 this time, even though downloading step was still slow.

After I reinstalled arrow, cmake  aligned to system installed arrow successfully.
